packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {FunctionComponent, useCallback, useEffect, useMemo, useRef} from 'react';
+
+import {
+    PaymentMethodProps,
+    PaymentMethodResolveId,
+    toResolvableComponent,
+} from '@bigcommerce/checkout/payment-integration-api';
+import { DynamicFormField, DynamicFormFieldType, FormContext } from "@bigcommerce/checkout/ui";
+import { FormField } from '@bigcommerce/checkout-sdk';
+import getPaypalCommerceRatePayValidationSchema from './validation-schemas/getPaypalCommerceRatePayValidationSchema';
+
+interface RatePayFieldValues {
+    ratepayBirthDate: string;
+    ratepayPhoneCountryCode: string;
+    ratepayPhoneNumber: string;
+}
+
+const formFieldData: FormField[] = [
+    {
+        name: 'ratepayBirthDate',
+        custom: false,
+        id: 'ratepayBirthDate',
+        label: 'payment.ratepay.birth_date',
+        required: true,
+        fieldType: DynamicFormFieldType.DATE,
+    },
+    {
+        name: 'ratepayPhoneCountryCode',
+        custom: false,
+        id: 'ratepayPhoneCountryCode',
+        label: 'payment.ratepay.phone_country_code',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        type: 'string',
+        maxLength: 2,
+    },
+    {
+        name: 'ratepayPhoneNumber',
+        custom: false,
+        id: 'ratepayPhoneNumber',
+        label: 'payment.ratepay.phone_number',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        maxLength: 11,
+        min: 8,
+    }
+];
+
+const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
+    method,
+    checkoutService,
+    paymentForm,
+    onUnhandledError,
+    paymentForm: {
+        isSubmitted,
+        setFieldValue,
+        setValidationSchema,
+        setSubmitted,
+    },
+    language,
+}) => {
+    const fieldsValues = useRef<Partial<RatePayFieldValues>>({});
```

Comment:
Maybe we should add logic for fixing bug with the Store Credit as I made it in this PR?
https://github.com/bigcommerce/checkout-js/pull/1349/commits/5f2c87796b537124cb1429f0211bfe4cbfd824a6#diff-581981c492a6d210cf0515845a7a2baff78ed352a93676dce8b838933d25c9d4

Reply:
Added fix